### DISCUSSION TITLE
Point light shadows, Area Lights?

### DISCUSSION THREAD

**Discussion by alexgu754** [google/filament#5090] [Ideas]
Is there an eta on area lights? or point light shadows
especially point light shadows. that seems like a fairly essential feature of any pbr engine

**Comment by pixelflinger** [comment]
No ETA. This has not been a priority for us.

The main technical issue is to figure out how to store the cubemaps (we don't have enough samplers to actually use a cubemap per point-light, and cubemap arrays are not supported on all hardware), and filter or blur them (for VSM) efficiently.

**Reply by TheOriginalBytePlayer** [reply]
Could you not have it work for some hardware and not for others?

**Reply by akshay-varshney** [reply]
@pixelflinger : Sorry just wanted to confirm one point here, does it mean that shadows work only for directional light not for point light? is it true?

**Reply by romainguy** [reply]
Spot lights can have shadows. Point lights cannot have shadows at this time.

**Comment by pixelflinger** [comment]
The only way you sort of can emulate a point light shadow is by using 6 spotlights with an opening of 90deg. not great.
